src/version.rs

```diff
@@ -17,31 +17,47 @@ use rdkafka::util::get_rdkafka_version;
 /// Version of the tremor crate;
 pub const VERSION: &str = env!("CARGO_PKG_VERSION");
 
-#[cfg(not(debug_assertions))]
-/// Long version include build info
-pub const VERSION_LONG: &str = env!("CARGO_PKG_VERSION");
-#[cfg(debug_assertions)]
-/// Long version include build info
-pub const VERSION_LONG: &str = concat!(env!("CARGO_PKG_VERSION"), " (DEBUG)");
-
 #[cfg(not(debug_assertions))]
 /// Checks if a we are in a debug build
 pub const DEBUG: bool = false;
 #[cfg(debug_assertions)]
 /// Checks if a we are in a debug build
 pub const DEBUG: bool = true;
 
+#[must_use]
+/// Provides formatting for "long" version name of build
+pub fn long_ver() -> String {
+    #[cfg(not(debug_assertions))]
+    const VERSION_LONG: &str = env!("CARGO_PKG_VERSION");
+    #[cfg(debug_assertions)]
+    const VERSION_LONG: &str = concat!(env!("CARGO_PKG_VERSION"), " (DEBUG)");
+    return match option_env!("VERSION_BRANCH") {
```

Comment:
This could be un-nested to:

```
    return match option_env!("VERSION_BRANCH") {
        Some("main") => VERSION_LONG.to_string(),
        Some(branch) => {
              // additional version info otherwise
              let commit_hash: &str = option_env!("VERSION_HASH").map_or("", |hash| hash);
              format!("{} {}:{}", VERSION_LONG, branch, commit_hash)
        }
        None => VERSION_LONG.to_string(), // default if option env not set
    };
```

Reply:
ah, clever. Learning Rust and appreciate the guidance :)